### DISCUSSION TITLE
v1.86.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#5025] [Announcements]
# v1.86.0

> [!warning]
> **Breaking Changes**
> - The server and mobile app **must** be on the same version for the application to work correctly.
> - The mobile app might log you out. You will need to log in again for this release since we are changing some of the APIs and cannot auto-resolve when the user is logged in. 
> - Docker compose auto-names resources based on the _project name_, with a fallback to the current folder. In #4906 we now set the project name to “immich” in our docker compose file. For existing installs, we recommend ignoring this change as it would lead to orphaned resources.
> 
> We are sorry for the inconvenience

## Highlights

Welcome to release `v1.86.0` of Immich. This release brings you some of the highly requested features and a lot of bug fixes and improvements. Let’s dive right into the highlighted features below:

- Partner sharing on the main timeline (Web & Mobile)
- New map tile server
- Support for additional facial recognition models
- User’s avatar color
- Notable fix: “Missing” for the recognize faces job now will only process an asset a single time, even when no faces are detected (#4854)
- Notable fix: user first/last name has been merged into a single “name” field (#4915)

### Partner sharing on the main timeline 

In this release we have added an option to enable parter-shared photos to be displayed in the main timeline. This feature can be enabled on a per-partner basis and can be viewed and updated on both the web and mobile app.

For the web interface: `Account’s Setting > Sharing > Show in timeline`

![image](https://github.com/immich-app/immich/assets/27055614/74430223-0ea4-485e-ba72-f92cd0695492)

For the mobile app: from the partner’s view, on the top right corner of the app bar

<p align="center">
<img src="https://github.com/immich-app/immich/assets/10599762/6587877e-e22e-4380-9197-86cf36d59c56" width="250" > 
<img src="https://github.com/immich-app/immich/assets/10599762/f4a7a45e-02fd-4235-89a0-f448dc53bd66" width="250" > 
</p>

Related to this new feature, the people-sharing card has been redesigned to make it easier to understand what is being shared, and with whom.

Now that assets from multiple users can appear in the timeline, there are some caveats, namely:

- You can view partners’ media and add them to an album. No other actions are allowed on those assets.
- Facial data is not shared (yet)
- The search result doesn’t include partners’ assets (yet)
- You cannot favorite partners’ assets (yet)
- You are at the mercy of the partners' actions (as it should be) on the asset they own. If the partner decides to delete or archive an asset belonging to them, it will not be displayed on your timeline.
- Partner sharing may result in displaying duplicate assets on the main timeline. We would like to address this soon.

### New map tile server

Since Immich has been growing quickly, we got a reminder from the community that we were using OpenStreetMap’s tile server. We have switched to a different default tile server to avoid violating the [Tile Usage Policy](https://operations.osmfoundation.org/policies/tiles/). This created an opportunity to build our custom layer to make the map’s look-and-feel slicker.

![Light Map](https://github.com/immich-app/immich/assets/27055614/396ae7ab-aeeb-4ca0-8bea-54b932beb066)
![Dark Map](https://github.com/immich-app/immich/assets/27055614/324d600e-a592-4c51-a2b9-3a9af45fbc93)

### Support for additional facial recognition models

We’ve added additional support for `antelopev2` and `buffalo_m` as the supported facial recognition models:
 
`antelopev2` has the same detection model as `buffalo_l`, but a roughly 50% larger recognition model. 
`buffalo_m` has the same recognition model as `buffalo_l`, but a smaller detection model.

### User’s avatar color

We added an option to change the color of your avatar if you don’t want to upload a profile picture. This mechanism will help to distinguish the user in the Activity panel on the album page.

https://github.com/immich-app/immich/assets/74269598/2a133113-190c-49b6-8dea-2ddf113507c8
- - - -

And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 1FvEp6P6NM8EZEkpGUFAN2LqJ1gxusNxZX

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### Mobile
* feat(mobile): unify partner assets on timeline by @fyfrey in https://github.com/immich-app/immich/pull/4974
* feat(mobile): Added fr-CA translations by @opdelta in https://github.com/immich-app/immich/pull/5023
* fix(mobile): display simple album date when the range is single day by @adamantike in https://github.com/immich-app/immich/pull/4919
* fix(mobile): add label for expire after in shared link edit page by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4920
* fix(mobile): fix JSON-format typos in Asset model by @aphymi in https://github.com/immich-app/immich/pull/4942
* fix(mobile): run user sync operation with lock by @fyfrey in https://github.com/immich-app/immich/pull/4984
* fix(mobile): better icon spacing in the bottom bar by @fyfrey in https://github.com/immich-app/immich/pull/5010
* fix(mobile): better app state handling by @fyfrey in https://github.com/immich-app/immich/pull/4989
* fix(mobile): freeze at splash screen when updating from 1.85 to the new version by @alextran1502 in https://github.com/immich-app/immich/pull/5012
* refactor(mobile): build context extensions by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4923
* chore(mobile): Translations update by @alextran1502 in https://github.com/immich-app/immich/pull/5011

### Server
* feat(server): don't re-run face recognition on assets without any faces by @sushain97 in https://github.com/immich-app/immich/pull/4854
* fix(server): non-admin cannot use map by @alextran1502 in https://github.com/immich-app/immich/pull/4934
* fix(server): replace first and last name with single field by @brianjaustin in https://github.com/immich-app/immich/pull/4915
* refactor(server): auth dtos by @jrasm91 in https://github.com/immich-app/immich/pull/4881
* chore(server): server-config e2e test by @jrasm91 in https://github.com/immich-app/immich/pull/4941

### Web
* feat(web): Maplibre by @danieldietzler in https://github.com/immich-app/immich/pull/4294
* feat(web): show partners assets on the main timeline by @alextran1502 in https://github.com/immich-app/immich/pull/4933
* feat(web): Add file path info for owned assets by @dpantel in https://github.com/immich-app/immich/pull/4943
* feat(web): use css vraiables by @jrasm91 in https://github.com/immich-app/immich/pull/5022
* feat(web): uniform random asset selection in slideshow mode by @brighteyed in https://github.com/immich-app/immich/pull/4953
* feat(web): user avatar color by @martabal in https://github.com/immich-app/immich/pull/4779
* fix(web): like in global activity by @martabal in https://github.com/immich-app/immich/pull/4916
* fix(web): maplibre prod build by @danieldietzler in https://github.com/immich-app/immich/pull/4977
* fix(web): better year labels by @waclaw66 in https://github.com/immich-app/immich/pull/5009
* refactor(web): map by @danieldietzler in https://github.com/immich-app/immich/pull/4957

### Machine Learning
* feat(ml): add face models by @mertalev in https://github.com/immich-app/immich/pull/4952
* chore(ml): use strict mypy by @mertalev in https://github.com/immich-app/immich/pull/5001
* fix(ml): fix type checks in tests by @mertalev in https://github.com/immich-app/immich/pull/5017

### Documentation and CI
* docs: Typo in libraries exclusion pattern by @Ploonet in https://github.com/immich-app/immich/pull/4972
* chore!: add docker project name by @Mansour-J in https://github.com/immich-app/immich/pull/4906
* chore: re-enable renovate by @bo0tzz in https://github.com/immich-app/immich/pull/4955
* chore: update Makefile to use docker compose by @alextran1502 in https://github.com/immich-app/immich/pull/4967
* chore(renovate): PR exiftool updates separately by @bo0tzz in https://github.com/immich-app/immich/pull/4985
* chore(ci): use custom base for server image build by @bo0tzz in https://github.com/immich-app/immich/pull/4456

### Dependencies Updates
* fix(deps): update server by @renovate in https://github.com/immich-app/immich/pull/4982
* fix(deps): update dependency tailwindcss to v3.3.5 by @renovate in https://github.com/immich-app/immich/pull/4980
* fix(deps): update dependency systeminformation to v5.21.17 by @renovate in https://github.com/immich-app/immich/pull/4979
* fix(deps): update dependency yaml to v2.3.4 by @renovate in https://github.com/immich-app/immich/pull/4981
* chore(renovate): Ignore un-bumpable mobile deps by @bo0tzz in https://github.com/immich-app/immich/pull/5007
* chore(deps): update ghcr.io/nginxinc/nginx-unprivileged:1.25.0-alpine3.17 docker digest to 5ebb90a by @renovate in https://github.com/immich-app/immich/pull/4960
* chore(deps): update redis:6.2-alpine docker digest to 3995fe6 by @renovate in https://github.com/immich-app/immich/pull/4964
* chore(deps): update postgres:14-alpine docker digest to 874f566 by @renovate in https://github.com/immich-app/immich/pull/4963
* chore(deps): update dependency @types/byte-size to v8.1.2 by @renovate in https://github.com/immich-app/immich/pull/4965
* chore(deps): update dependency @types/chai to v4.3.10 by @renovate in https://github.com/immich-app/immich/pull/4966
* chore(deps): update dependency @types/jest to v29.5.8 by @renovate in https://github.com/immich-app/immich/pull/4969
* chore(deps): update dependency @types/cli-progress to v3.11.5 by @renovate in https://github.com/immich-app/immich/pull/4968
* chore(deps): update dependency @types/mock-fs to v4.13.4 by @renovate in https://github.com/immich-app/immich/pull/4975
* chore(deps): update dependency jest-extended to v4.0.2 by @renovate in https://github.com/immich-app/immich/pull/4976
* chore(deps): update dependency @types/mime-types to v2.1.4 by @renovate in https://github.com/immich-app/immich/pull/4971
* chore(deps): update dependency @types/js-yaml to v4.0.9 by @renovate in https://github.com/immich-app/immich/pull/4970
* chore(deps): update dependency @types/node to v20.9.0 by @renovate in https://github.com/immich-app/immich/pull/4983
* chore(deps): update ghcr.io/nginxinc/nginx-unprivileged docker tag to v1.25.1 by @renovate in https://github.com/immich-app/immich/pull/4978
* chore(deps): update dependency eslint-plugin-jest to v27.6.0 by @renovate in https://github.com/immich-app/immich/pull/4987
* chore(deps): update dependency eslint to v8.53.0 by @renovate in https://github.com/immich-app/immich/pull/4986
* chore(deps): update dependency @types/jest to v29.5.8 by @renovate in https://github.com/immich-app/immich/pull/4995

## New Contributors
* @sushain97 made their first contribution in https://github.com/immich-app/immich/pull/4854
* @Mansour-J made their first contribution in https://github.com/immich-app/immich/pull/4906
* @aphymi made their first contribution in https://github.com/immich-app/immich/pull/4942
* @brianjaustin made their first contribution in https://github.com/immich-app/immich/pull/4915
* @dpantel made their first contribution in https://github.com/immich-app/immich/pull/4943
* @opdelta made their first contribution in https://github.com/immich-app/immich/pull/5023

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.85.0...v1.86.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.86.0'>v1.86.0</a>.</em>

**Comment by iheartpizza1** [comment]
Hi, I've encountered several issues upgrading to this release, which were not mentioned in the breaking changes list.

1. I'm using a `config.json` file. When starting the containers, `immich-server & microservices` threw errors "Unknown keys in json [object Object]; invalid config file". After figuring out what has changed in the config file using diff of v1.85.0, I've changed my config file and tried starting the containers again.
`[Nest] 7  - 11/14/2023, 8:45:53 AM   ERROR [SystemConfigCore] Unable to load configuration file: /usr/immich-config.json due to Error: Unknown keys found: [object Object]
Error: Unknown keys found: [object Object]
    at SystemConfigCore.loadFromFile (/usr/src/app/dist/domain/system-config/system-config.core.js:313:27)
    at async SystemConfigCore.getConfig (/usr/src/app/dist/domain/system-config/system-config.core.js:241:44)
    at async SystemConfigService.getConfig (/usr/src/app/dist/domain/system-config/system-config.service.js:38:24)
    at async Injector.instantiateClass (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:369:37)
    at async callback (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:65:34)
    at async Injector.resolveConstructorParams (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:144:24)
    at async Injector.loadInstance (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:70:13)
    at async Injector.loadProvider (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:97:9)
    at async /usr/src/app/node_modules/@nestjs/core/injector/instance-loader.js:56:13
    at async Promise.all (index 24)
[Nest] 7  - 11/14/2023, 8:45:53 AM   ERROR [ExceptionHandler] Invalid configuration file
Error: Invalid configuration file
    at SystemConfigCore.loadFromFile (/usr/src/app/dist/domain/system-config/system-config.core.js:319:23)
    at async SystemConfigCore.getConfig (/usr/src/app/dist/domain/system-config/system-config.core.js:241:44)
    at async SystemConfigService.getConfig (/usr/src/app/dist/domain/system-config/system-config.service.js:38:24)
    at async Injector.instantiateClass (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:369:37)
    at async callback (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:65:34)
    at async Injector.resolveConstructorParams (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:144:24)
    at async Injector.loadInstance (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:70:13)
    at async Injector.loadProvider (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:97:9)
    at async /usr/src/app/node_modules/@nestjs/core/injector/instance-loader.js:56:13
    at async Promise.all (index 24)`
Referring to the _removed_ config key `map.tileUrl`, and the _added_ keys `map.lightStyle, map.darkStyle`.

2. I've tried unbinding the config file from the server & microservices containers, but that caused them to throw errors "invalid config file, cannot find config file at <previous, non-existent path>". So after starting using the config file, we're stuck using it without the ability to stop using it?
`[Nest] 7  - 11/14/2023, 8:48:08 AM   ERROR [SystemConfigCore] Unable to load configuration file: /usr/immich-config.json due to Error: ENOENT: no such file or directory, open '/usr/immich-config.json'
Error: ENOENT: no such file or directory, open '/usr/immich-config.json'
[Nest] 7  - 11/14/2023, 8:48:08 AM   ERROR [ExceptionHandler] Invalid configuration file
Error: Invalid configuration file
    at SystemConfigCore.loadFromFile (/usr/src/app/dist/domain/system-config/system-config.core.js:319:23)
    at async SystemConfigCore.getConfig (/usr/src/app/dist/domain/system-config/system-config.core.js:241:44)
    at async SystemConfigService.getConfig (/usr/src/app/dist/domain/system-config/system-config.service.js:38:24)
    at async Injector.instantiateClass (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:369:37)
    at async callback (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:65:34)
    at async Injector.resolveConstructorParams (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:144:24)
    at async Injector.loadInstance (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:70:13)
    at async Injector.loadProvider (/usr/src/app/node_modules/@nestjs/core/injector/injector.js:97:9)
    at async /usr/src/app/node_modules/@nestjs/core/injector/instance-loader.js:56:13
    at async Promise.all (index 24)`
    
3. Cannot login from Android app, getting some error "url /oauth/config not found". Downgrading to v1.85.0 didn't help, as now when trying to login from the app, I'm getting server error "QueryFailedError: column user.firstName does not exist". After upgrading again to v1.86.0, and again changing the config file, suddenly the app is able to login.

**Reply by jrasm91** [reply]
Yes we did not document the config change very well for the tile url.

I would recommend deleting all config that you are not specifically changing, then it will use the default values and you will have less breaking changes to fix.

**Comment by maltokyo** [comment]
https://github.com/immich-app/immich/discussions/5028
Any advice is welcome

**Comment by sean13013** [comment]
The "partner sharing on main timeline" is awesome!

**Comment by daniele-athome** [comment]
I previously configured a custom tile server URL, but I see the setting has disappeared.  What server is Immich using now? Can I still use a custom URL?

**Reply by bo0tzz** [reply]
We've switched to using Maplibre-gl and vector tiles. You can still configure a custom URL, though now it needs to point to a [MapLibre style specification](https://maplibre.org/maplibre-style-spec/) json file. AFAIK most map APIs offer those.

The new default tile server is sponsored and hosted by some friends of the maintainer team.

**Reply by daniele-athome** [reply]
Oh, sorry I thought "the style URL" was just something to add only custom CSS to the map. So it's also used the get the actual tiles, ok.
Thank you.

**Comment by pmontgo33** [comment]
Do you know when the 1.86 will come to F-Droid? I am a new user, and couldn't figure out why I can't login to the mobile app until I found these release notes. Thanks and looking forward to trying this out!

**Reply by shenlong-tanwen** [reply]
It usually takes about 4-7 days for the app to be updated on F-Droid.
